### PR TITLE
Docker Documentation

### PR DESCRIPTION
**Pull request by 1000TurquoisePogs**
With Docker support coming soon, I'm starting to edit the documentation to find places where the story branches between what you do when you are using docker versus what you do when you are not using docker. The other approach would have been to make a completely separate document, but the code is largely the same so there would be quite a bit of duplication.

It seems to me just the getting started and user guide sections need revision for docker, possibly troubleshooting as we learn more about common issues.

Signed-off-by: 1000TurquoisePogs <sgrady@rocketsoftware.com>

### CODE REVIEW

File: docs/getting-started/freqaskques.md [hunk 1]
```diff
@@ -97,7 +97,21 @@ To get up and running with the Zowe CLI component quickly, see [Zowe CLI quick s
 
 <summary></summary>
 
-The primary prerequisites is Java on z/OS and the z/OS Management Facility enabled and configured. For a complete list of software requirements listed by component, see [System requirements](../user-guide/systemrequirements.md).
+Prerequisites vary by component used, but in most cases the primary prerequisites are Java and NodeJS on z/OS and the z/OS Management Facility enabled and configured. For a complete list of software requirements listed by component, see [System requirements](../user-guide/systemrequirements.md).
```
Comment (ajinrocket) [inline docs/getting-started/freqaskques.md:100]: Users must have Java and NodeJS enabled and configured on z/OS and z/OS management facility. 
**Note**: Prerequisites can vary based on what components are being used.

File: docs/getting-started/freqaskques.md [hunk 2]
```diff
@@ -97,7 +97,21 @@ To get up and running with the Zowe CLI component quickly, see [Zowe CLI quick s
...
+
+</details>
+
+### What's the difference between using Zowe with or without Docker?
+
+<details class="zowe-faq">
+
+<summary></summary>
+
+There are multiple download options for Zowe server components, such as PAX and SMPE. Docker is another option, which allows you to run some Zowe server components outside of z/OS.
```
Comment (ajinrocket) [inline docs/getting-started/freqaskques.md:110]: Docker is a download option for Zowe that allows you to run certain Zowe server components outside of z/OS.

File: docs/getting-started/freqaskques.md [hunk 3]
```diff
@@ -97,7 +97,21 @@ To get up and running with the Zowe CLI component quickly, see [Zowe CLI quick s
...
+This may be helpful for those who don't have adequate access to a z/OS machine, or who have extra capacity off-Z that they want to utilize.
```
Comment (ajinrocket) [inline docs/getting-started/freqaskques.md:111]: This option is useful for users who do not have adequate access to a z/OS machine, as well as users who have extra capacity off-Z that they want to utilize.

File: docs/getting-started/freqaskques.md [hunk 4]
```diff
@@ -97,7 +97,21 @@ To get up and running with the Zowe CLI component quickly, see [Zowe CLI quick s
 
 <summary></summary>
 
-The primary prerequisites is Java on z/OS and the z/OS Management Facility enabled and configured. For a complete list of software requirements listed by component, see [System requirements](../user-guide/systemrequirements.md).
+Prerequisites vary by component used, but in most cases the primary prerequisites are Java and NodeJS on z/OS and the z/OS Management Facility enabled and configured. For a complete list of software requirements listed by component, see [System requirements](../user-guide/systemrequirements.md).
+
+</details>
+
+### What's the difference between using Zowe with or without Docker?
```
Comment (ajinrocket) [inline docs/getting-started/freqaskques.md:104]: We are eventually going to want to link to a section in the documentation regarding Docker (once it's written).

File: docs/getting-started/freqaskques.md (hunk 1 above)
Comment (1000TurquoisePogs) [reply docs/getting-started/freqaskques.md:100]: It's not a "must" if Docker is used, that's why I changed it.

**Comment by 1000TurquoisePogs** [comment]
> Note: the Zowe CLI can also be run in a Docker container. It's a bit different bc you're running the CLI in a container on the client side instead of running our server components in a Docker container on mainframe.
> 
> For example, an automated pipeline (like Jenkins) might deploy Docker containers that run the CLI to do some build or testing tasks.
> 
> Might be worth a mention when talking about Zowe + Docker

I added an FAQ section that attempts to resolve confusion on the topic. All of the install & config topics I did modify later on in the docs are about the "zowe runtime" or more specifically the server runtime. I hope that pre-existing terminology in the documentation makes the difference between this and the CLI clear. Over time if the CLI does get distributed within the same docker image, or as another docker image option, the documentation will need to remind & clarify the difference.

**Comment by nannanli** [comment]
One broken link is reported:

File: docs/user-guide/systemrequirements.md [hunk 5]
```diff
@@ -81,13 +91,41 @@ See [Reviewing the instance env file](../user-guide/configure-instance-directory
...
+Once you have Docker, the Zowe Docker image has the following requirements
 
-To enable multi-factor authentication, you must install [IBM Z Multi-Factor Authentication](https://www.ibm.com/us-en/marketplace/ibm-multifactor-authentication-for-zos). For information on using MFA in Zowe, see [Multi-Factor Authentication](mvd-configuration.md#multi-factor-authentication-configuration).
+- 4 GB free RAM, 8 GB recommended
+- 4 GB free disk space
+- Network access to the z/OS host. The Linux host must be able to communicate with the z/OS host.
 
-### Using web tokens for SSO on ZLUX and ZSS
+When using Docker, z/OS is still required and if using the Application Framework or ZSS, installing ZSS on the z/OS host is still required. See these requirements:
+- [z/OS system requirements](#zos-system-requirements-host)
```
Comment (1000TurquoisePogs) [inline docs/user-guide/systemrequirements.md:108]: Thanks, just fixed the broken link which was here.

**Comment by nannanli** [comment]
@1000TurquoisePogs Hi Sean, I spotted and fixed several other minor issues (links, typos, missing stops, etc). Made those changes to your branch directly. See [my commit](https://github.com/zowe/docs-site/pull/1398/commits/c4c58608ad0f428767b60f71c4b34d4337319b93).

**Comment by nannanli** [comment]
@1000TurquoisePogs Could you also upload the source files for the following graphics? Thanks! 

- docs/images/common/zowe-architecture-docker.png
- docs/images/common/zowe-architecture.png
- docs/images/common/zowe-docker-install-diagram.png

**Pull request merged**